VMBackup/main/Utils/HandlerUtil.py

```diff
@@ -244,10 +247,73 @@ def save_seq(self):
     def set_last_seq(self,seq):
         waagent.SetFileContents('mrseq', str(seq))
 
+    def get_machine_id(self):
+        machine_id_file = "/etc/azure/machine_identity_FD76C85E-406F-4CFA-8EB0-CF18B123358B"
+        machine_id = ""
+        try:
+            if os.path.exists(os.path.dirname(machine_id_file)):
+                file_pointer = open(machine_id_file, "r")
+                machine_id = file_pointer.readline()
+                file_pointer.close()
+            else:
+                mi = MachineIdentity()
+                machine_id = mi.stored_identity()[1:-1]
+                os.makedirs(os.path.dirname(machine_id_file))
+                file_pointer = open(machine_id_file, "w")
+                file_pointer.write(machine_id)
+                file_pointer.close()
+        except:
+            errMsg = 'Failed to retrieve the unique machine id with error: %s, stack trace: %s' % (str(e), traceback.format_exc())
+            self.log(errMsg, False, 'Error')
+ 
+        self.log("Unique Machine Id  : {0}".format(machine_id))
+        return machine_id
+
+    def get_total_used_size(self):
+        try:
+            df = subprocess.Popen(["df"], stdout=subprocess.PIPE)
+            '''
+            Sample output of the df command
+
+            Filesystem     1K-blocks    Used Available Use% Mounted on
+            udev             1756684      12   1756672   1% /dev
+            tmpfs             352312     420    351892   1% /run
+            /dev/sda1       30202916 2598292  26338592   9% /
+            none                   4       0         4   0% /sys/fs/cgroup
+            none                5120       0      5120   0% /run/lock
+            none             1761552       0   1761552   0% /run/shm
+            none              102400       0    102400   0% /run/user
+            none                  64       0        64   0% /etc/network/interfaces.dynamic.d
+            tmpfs                  4       4         0 100% /etc/ruxitagentproc
+            /dev/sdb1        7092664   16120   6693216   1% /mnt
+
+            '''
+            output = df.communicate()[0]
```

Comment:
check if communicate method is required or the same can be done by using poll/wait followed by df.stdout